my-code/whats-in/src/App.jsx

```diff
@@ -0,0 +1,90 @@
+import { Routes, Route, useNavigate } from 'react-router-dom';
+import React, { useEffect, useState } from 'react'
+import Homepage from './pages/Homepage'
+import Moviepage from './pages/Movie'
+import Header from './components/Header'
+
+function App() {
+  const [searchText, setSearchText] = useState('')
+  const [movies, setMovies] = useState()
+  const [selectedMovie, setSelectedMovie] = useState(null)
+  const [selectedMovieInfo, setSelectedMovieInfo] = useState()
+  const [isLoading, setIsLoading] = useState(false)
+  const [likedMovies, setLikedMovies] = useState([]);
+  const [isEmptySearch, setIsEmptySearch] = useState(true)
+  const navigate = useNavigate()
+  
+  const handleSearch = (text) => {
+    setSearchText(text)
+  }
+
+  useEffect(() => {
+    fetch(`https://www.omdbapi.com/?s=${searchText}&apikey=${process.env.REACT_APP_API_KEY}`)
+    .then(response => response.json())
+    .then(data => {
+      if (searchText === '') {
+        setIsEmptySearch(true)
+        setMovies('')
+      } else {
+        setIsEmptySearch(false)
+        setMovies(data.Search)
+      }
+    })
+    .catch(error => {
+      console.error(error);
+    })
+  }, [searchText])
+
+  const handleMovieSelection = (movie) => {
+    setSelectedMovie(movie)
+  }
+
+  useEffect(() => {
+    if (selectedMovie && selectedMovie.imdbID) {
+      setIsLoading(true)
```

Comment:
the name of this variable is a bit questionable:
given you have two separate network calls within this component, how do we know to which the loading refers to?
You could also have a loading indicator for when the user is searching something, so it would make sense to name this `loadingDetails` or similar